### ISSUE TITLE
[Bug] Google.com is not setting title on first page load

### ISSUE THREAD

**Issue by boek** [mozilla-mobile/fenix#6826]
## Steps to reproduce
Go to google.com

### Expected behavior
Title is shown on the tab while looking at the tab switcher

### Actual behavior
Title area is empty

Also reproducible on the reference browser

**Comment by sblatz** [comment]
I just reproduced:
1. New tab
2. Search with google
3. Title does not change
4. Shortcuts > DuckDuckGo
5. Make a search
6. Title does not change


It seems like we're not getting an `onTitleChanged` callback in some cases when searching, so probably AC related.

**Comment by darkwing** [comment]
@csadilek , @grigoryk Any thoughts on what could cause this issue?

**Comment by boek** [comment]
@darkwing so I just checked and this isn't happening in the reference browser anymore. There must be something that we're not doing correctly

**Comment by darkwing** [comment]
Odd.  I threw a breakpoint into `Session.kt`'s:

```
var title: String by Delegates.observable("") { _, old, new ->
        if (notifyObservers(old, new) { onTitleChanged(this@Session, new) }) {
            store?.syncDispatch(UpdateTitleAction(id, new))
        }
    }
```

... and it hits twice when going to google.com:

![Gone](https://user-images.githubusercontent.com/46655/74293308-06f0bb00-4d00-11ea-9cbf-47239d82fb9f.gif)

It's there and then it's gone.  Will continue trying to figure this out.  The reference browser wont build for me and the codebase has no `onTitleChanged` actions so I"m not sure what they could be doing differently.

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on the latest **`Nightly Build #20550606 from 02/24`** using the following devices:
• Google Pixel 3a XL (Android 10)
• Google Pixel 3a (Android 9)
• Huawei Mate 20 Lite (Android 8.1.0)
• Samsung Galaxy S7 (Android 7)
• OnePlus A3 (Android 6.0.1)
• LG Nexus 4 (Android 5.1.1)

► Video
![20200224-101223](https://user-images.githubusercontent.com/51314259/75136860-a5acec80-56ee-11ea-9fa1-79fb8a943312.gif)
